coffea/jetmet_tools/JetCorrectionUncertainty.py

```diff
@@ -150,11 +151,19 @@ def getUncertainty(self, **kwargs):
             #in a zip iterator
 
         """
+        cache = kwargs.pop('lazy_cache', None)
         uncs = []
         for i, func in enumerate(self._funcs):
             sig = func.signature
-            args = []
-            for input in sig:
-                args.append(kwargs[input])
-            uncs.append(func(*tuple(args)))
+            args = tuple(kwargs[input] for input in sig)
+
+            if isinstance(args[0], awkward.array.base.AwkwardArray):
+                uncs.append(awkward.VirtualArray(func, args=args))
+            elif isinstance(args[0], np.ndarray):
+                uncs.append(func(*args))  # np is non-lazy
+            elif isinstance(args[0], awkward1.highlevel.Array):
+                uncs.append(awkward1.virtual(func, args=args, length=len(args[0]), cache=cache))
```

Comment:
You should provide a `form` to ak1 virtual so that it knows what is in the content. In this case, it is just a number right? You can use a simple dtype as the form argument in that case

Reply:
only len is needed here.